From 8a334083d5243c5e4010289687651e1b70548535 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Thu, 26 Feb 2015 16:36:11 +0100
Subject: [PATCH 0370/1800] Fix a crash when 'Check for supported
 authentication types' clicked

The crash happened when there was an attempt to connect securely
to the server, but the server provided untrusted certificate,
thus Evolution wanted to ask for the certificate trust.
---
 mail/e-mail-ui-session.c | 21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/mail/e-mail-ui-session.c b/mail/e-mail-ui-session.c
index b5b02ffc1c..fd62835785 100644
--- a/mail/e-mail-ui-session.c
+++ b/mail/e-mail-ui-session.c
@@ -531,18 +531,6 @@ mail_ui_session_remove_service (CamelSession *session,
 	e_mail_account_store_remove_service (store, NULL, service);
 }
 
-CamelCertTrust
-e_mail_ui_session_trust_prompt (CamelSession *session,
-                                CamelService *service,
-                                GTlsCertificate *certificate,
-                                GTlsCertificateFlags errors)
-{
-	g_type_ensure (E_TYPE_MAIL_UI_SESSION);
-
-	return CAMEL_SESSION_CLASS (e_mail_ui_session_parent_class)->
-		trust_prompt (session, service, certificate, errors);
-}
-
 static CamelFilterDriver *
 mail_ui_session_get_filter_driver (CamelSession *session,
                                    const gchar *type,
@@ -701,6 +689,15 @@ mail_ui_session_trust_prompt (CamelSession *session,
 	return response;
 }
 
+CamelCertTrust
+e_mail_ui_session_trust_prompt (CamelSession *session,
+                                CamelService *service,
+                                GTlsCertificate *certificate,
+                                GTlsCertificateFlags errors)
+{
+	return mail_ui_session_trust_prompt (session, service, certificate, errors);
+}
+
 typedef struct _TryCredentialsData {
 	CamelService *service;
 	const gchar *mechanism;
-- 
2.15.0

